src/Analyzers/CSharp/Analyzers/ConvertProgram/ConvertProgramAnalysis_ProgramMain.cs

```diff
@@ -36,7 +36,7 @@ public static bool IsApplication(CompilationOptions options)
 
             // resiliency check for later on.  This shouldn't happen but we don't want to crash if we are in a weird
             // state where we have top level statements but no 'Program' type.
-            var programType = compilation.GetBestTypeByMetadataName(WellKnownMemberNames.TopLevelStatementsEntryPointTypeName);
+            var programType = compilation.GlobalNamespace.GetTypeMembers(WellKnownMemberNames.TopLevelStatementsEntryPointTypeName, arity: 0).FirstOrDefault();
```

Comment:
>GlobalNamespace.GetTypeMembers(WellKnownMemberNames.TopLevelStatementsEntryPointTypeName, arity: 0).FirstOrDefault(); [](http://example.com/codeflow?start=42&length=117)

What is the reason for this change?

Reply:
This is required to remain support for the following case:
```cs
System.Console.WriteLine("Hello World!");
file partial class Program { }
```
The analyzer previously relied on the ability to get `Program` type from compilation by metadata name. However, in the given case top-level type `Program` is file-scoped, meaning, that after changes in this PR it is no longer possible to get it this way. Therefore we now get it via `GetTypeMembers` API, which is not affected by the change.